content/en/threat-model.md

```diff
@@ -0,0 +1,123 @@
+
+In this threat model, we consider the compromise of any of the following:
+* Fulcio CA server
+* Fulcio CT Log
+* Rekor server
+* Fulcio monitor
+* Rekor monitor
+* OIDC server: OIDC issuer/IdP
+* OIDC account
+* TUF Root of trust
+
+Here’s a high-level diagram of the Sigstore signing flow, with the components and steps that might be compromised highlighted in red:
+
+![Sigstore signing flow](/sigstore-threat-model-signing.svg)
+
+
+## Main takeaways
+* Identity and consistency monitoring is critical to mitigate risks of compromise to both the Fulcio CT log and Rekor. All signers should monitor the Fulcio CT log for unauthorized use of their identity, and all verifiers should look for alerts from consistency monitors.
+* The Fulcio CA is a particularly important source of trust and must be hardened.
+* OIDC issuers are highly trusted in Sigstore, and only properly hardened OIDC issuers should be used.
+* OIDC account compromise is not handled by Sigstore, but we recommend that OIDC issuers provide revocation in the case of a compromised OIDC account. In addition, we recommend the use of 2FA on all OIDC accounts used with Sigstore in order to reduce the likelihood of a compromise. 
+* TUF root of trust provides strong revocation abilities for all services managed by Sigstore.
+
+## Sigstore threat model
+
+### Scenario for Table 1
+This table analyzes the attacker capabilities when the attacker compromises different Sigstore components. This table assumes that Fulcio is used to sign any software or artifacts. The storage mechanism (repository, etc) for artifacts and metadata are out of scope for this model. The table description assumes TUF is not in play, except in the root of trust defined by Sigstore. We assume that there are multiple monitors of Rekor and the Fulcio CT log who communicate with a gossip protocol. A compromise means that all monitors are compromised. The table is sorted by the impact of the attacker’s capabilities. Compromises which have different capabilities with the same impact are grouped together. The actors in this table are described below:
+
+### Action: Alice signs an artifact with Sigstore
+Alice sends a certificate request to the Fulcio CA with a generated public key. The Fulcio CA requests a token from the OIDC provider. This token is then sent to Alice to verify. The Fulcio CA generates and signs a precertificate, which is uploaded to the Fulcio CT log. The Fulcio CT log generates and signs a SCT (signed certificate timestamp), which is sent back to the Fulcio CA. The Fulcio CA combines the SCT with the precertificate to get the final certificate which is sent to Alice. Alice uses the key associated with the certificate to sign the artifact. She uploads the certificate and signature to Rekor, which returns an SET (signed entry timestamp) promising inclusion in the log at a particular time. The verifier can use this time to ensure the certificate was used while it was valid.
```

Comment:
The flow with the OIDC IdP is a bit backwards. Alice would authenticate herself against the IdP to get an OIDC ID token which is sent to Fulcio along with the certificate request. Fulcio would then verify the authenticity of the ID Token with the OIDC IdP, and then extract the subject from the ID token and use as the SAN in the certificate.